org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/general/completion/FilePathCompletionParticipant.java

```diff
@@ -0,0 +1,265 @@
+/*******************************************************************************
+* Copyright (c) 2019 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.lsp4xml.extensions.general.completion;
+
+import static org.eclipse.lsp4xml.utils.FilesUtils.cleanPathForWindows;
+import static org.eclipse.lsp4xml.utils.FilesUtils.getFilePathSlash;
+import static org.eclipse.lsp4xml.utils.FilesUtils.getNormalizedPath;
+import static org.eclipse.lsp4xml.utils.OSUtils.isWindows;
+import static org.eclipse.lsp4xml.utils.StringUtils.isEmpty;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.net.URI;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.eclipse.lsp4j.CompletionItem;
+import org.eclipse.lsp4j.CompletionItemKind;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextEdit;
+import org.eclipse.lsp4xml.commons.BadLocationException;
+import org.eclipse.lsp4xml.dom.DOMDocument;
+import org.eclipse.lsp4xml.services.extensions.CompletionParticipantAdapter;
+import org.eclipse.lsp4xml.services.extensions.ICompletionRequest;
+import org.eclipse.lsp4xml.services.extensions.ICompletionResponse;
+import org.eclipse.lsp4xml.settings.SharedSettings;
+import org.eclipse.lsp4xml.utils.CompletionSortTextHelper;
+import org.eclipse.lsp4xml.utils.FilesUtils;
+import org.eclipse.lsp4xml.utils.StringUtils;
+
+/**
+ * FilePathCompletionParticipant
+ */
+public class FilePathCompletionParticipant extends CompletionParticipantAdapter {
+
+	public static final String FILE_SCHEME = "file";
+
+
+	@Override
+	public void onTagOpen(ICompletionRequest completionRequest, ICompletionResponse completionResponse) throws Exception {
+
+	}
+
+	@Override
+	public void onXMLContent(ICompletionRequest request, ICompletionResponse response) throws Exception {
+
+	}
+
+	@Override
+	public void onAttributeName(boolean generateValue, Range fullRange, ICompletionRequest request,
+			ICompletionResponse response, SharedSettings settings) throws Exception {
+
+	}
+
+	@Override
+	public void onAttributeValue(String valuePrefix, Range fullRange, boolean addQuotes, ICompletionRequest request,
+			ICompletionResponse response, SharedSettings settings) throws Exception {
+
+		DOMDocument xmlDocument = request.getXMLDocument();
+		String text = xmlDocument.getText();
+
+		// Get full attribute value range
+		// + 1 since it includes the quotations
+		int documentStartOffset = xmlDocument.offsetAt(fullRange.getStart()) + 1;
+		
+		String fullAttributeValue = valuePrefix;
+		if (isEmpty(fullAttributeValue)) {
+			return;
```

Comment:
Coudn't consider that if value is empty, it's the same thing than '.' ?

Reply:
Do we want to have path completion for every attribute all the time?

I thought it would be better to trigger when they type '.'

Reply:
No but it works like this today no?

Reply:
@NikolasKomonen here a commons case that it doesn't work with XML Schema " `xsi:noNamespaceSchemaLocation` (see `|` where completion is triggered:

If I use `.` in the path:

```xml
<data xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="./|">
```
I have completion for xsd files.

![image](https://user-images.githubusercontent.com/1932211/59091380-79a6cc80-890f-11e9-8a1d-1dd2f603408b.png)

If I have an empty path:

```xml
<data xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="|">
```
it doesn't work. Generally we write xsd file references without `./`